python/cudf/cudf/core/scalar.py

```diff
@@ -114,49 +116,45 @@ def _device_value_to_host(self):
         self._host_value = self._device_value._to_host_scalar()
 
     def _preprocess_host_value(self, value, dtype):
-        if isinstance(dtype, Decimal64Dtype):
-            # TODO: Support coercion from decimal.Decimal to different dtype
-            # TODO: Support coercion from integer to Decimal64Dtype
-            raise NotImplementedError(
-                "dtype as cudf.Decimal64Dtype is not supported. Pass a "
-                "decimal.Decimal to construct a DecimalScalar."
-            )
-        if isinstance(value, decimal.Decimal) and dtype is not None:
-            raise TypeError(f"Can not coerce decimal to {dtype}")
-
-        value = to_cudf_compatible_scalar(value, dtype=dtype)
-        valid = not _is_null_host_scalar(value)
-
-        if isinstance(value, decimal.Decimal):
-            # 0.0042 -> Decimal64Dtype(2, 4)
-            dtype = Decimal64Dtype._from_decimal(value)
-
+        if value is NA and dtype is not None:
+            return NA, dtype
+        elif type(value) is np.bool_:
```

Comment:
```suggestion
        elif type(value) in (bool, np.bool_):
```
we might have to handle this too if we are checking for a boolean value.

Reply:
This is the case this is meant to handle:

```
>>> pa.scalar(True)
<pyarrow.BooleanScalar: True>
>>> pa.scalar(np.bool_(True))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/scalar.pxi", line 851, in pyarrow.lib.scalar
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Could not convert True with type numpy.bool_: tried to convert to boolean
```
I think this is a pyarrow bug.

Reply:
> I think this is a pyarrow bug.

Definitely a pyarrow bug, because they handle `np.bool` correctly:

```python
>>> import pyarrow as pa
>>> import numpy as np
>>> pa.scalar(True)
<pyarrow.BooleanScalar: True>
>>> pa.scalar(np.bool(True))
<stdin>:1: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
<pyarrow.BooleanScalar: True>
>>> pa.scalar(np.bool_(True))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/scalar.pxi", line 851, in pyarrow.lib.scalar
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Could not convert True with type numpy.bool_: tried to convert to boolean
```

Lets raise an issue with arrow and handle the `np.bool_` separately like:
```python
pa.scalar(bool(np.bool_(True)))
<pyarrow.BooleanScalar: True>
```